src/main/java/net/minecraftforge/snowblower/Generator.java

```diff
@@ -95,6 +101,16 @@ public Generator setup(String branchName, @Nullable String remoteName, Config cf
                 git.branchDelete().setBranchNames(branchName).setForce(true).call(); // Delete existing branch
                 exists = false;
                 git.checkout().setOrphan(true).setName(branchName).call(); // Move to correctly named branch
+            } else if (!fresh && this.checkout && this.remoteName != null && git.getRepository().resolve(this.remoteName + "/" + branchName) != null) {
+                if (exists) {
+                    if (branchName.equals(currentBranch)) {
+                        git.checkout().setOrphan(true).setName("orphan_temp").call();    // Move to temp branch so we can delete existing one
+                        temp = true;
+                    }
+                    git.branchDelete().setBranchNames(branchName).setForce(true).call(); // Delete branch
+                }
```

Comment:
This should probably be moved in a separate method to avoid code duplication.

Reply:
It's pretty short and sets a temp = true variable which must be used at the end of the method, so it is not an easy extraction.

Reply:
It could be extracted into a method returning a boolean.